verification-by-reference.md

```diff
@@ -0,0 +1,331 @@
+# Notary v2 - Verification  by Reference
+
+5/1/2020
+[Steve Lasker](https://github.com/SteveLasker)
+
+## Overview
+
+One of the many questions of Notary v2 is how will verification (signatures) be persisted within a registry. This paper outlines an option for persisting verification content as additional artifacts in the registry, with a reverse lookup. This would enable multiple verifications to exist without changing the original artifact, or the digest of the original content.
+
+## Terms & references
+
+For the purposes of this document, we defer what a verification means. A verification may be a signature, where an org, company or entity states it attests to the current state of an artifact. Or a larger verification document that has additional content to prove verification of an artifact. This doc experiments with the avoidance of signatures as the reference.
+
+## Goals
+
+1. An artifact and it’s digest shall not change as the result of a verification object being added.
+1. Verification steps along a workflow must not be required to change how they reference the target artifact. A kube deployment yaml, helm chart, or docker compose file must not be required to change their tag reference of ubuntu:1404 or ubuntu@sha256:2468 digest reference.
+1. A collection of verification objects may be associated with an artifact.
+1. Leverage the garbage collection infrastructure registry operators have implemented. 
+Garbage collection represents significant investments for registry services which are based on OCI Manifest and OCI Index. We should aim to utilize these existing schemas, or only slight verifications from them to maximize the opportunity for registry implementations to adopt Notary v2.
+1. Minimize requirements to change persistence and object stores used by registry operators. Similar to garbage collection, we should work within the constraints of manifest, index and layers to represent verification objects.
+
+## Non-goals
+
+1. Work within the constraints of the existing OCI-distribution spec api set. 
+Supporting Notary v1, or Docker Content Trust, requires new APIs. APIs are comparatively cheap to implement atop the persistence stores of registries. 
+New APIs should be part of the Notary v2 spec, which may represent changes to the OCI-distribution spec or implementations of a new extension model over the Distribution spec.
+1. Compatibility with Notary v1. Registries that have implemented Notary v1 are looking for a better solution. We cumulatively have little existing usage, and if successful, we expect all customers would rapidly move to Notary v2 eliminating the need to maintain two sets of APIs for an extended period of time.
```

Comment:
I love that this is explicitly called out, but as in implementor (and user) of notary-v1, I don't see v2 as solving any of my issues. To wit, v1 has very rough tooling, and is complex. v2, adds significant complexity.

Reply:
thanks @directionless, I don't think we've gotten far enough along with v2 to say we've added complexity. Our primary goal is to balance complexity with usability. I'd actually say I'm confident we'll tame the complexity beast, as if we don't we won't ship. As we've seen, if it's too complex to use, customer wont. We need a usable solution.